### PR TITLE
Added support for Java 11

### PR DESCRIPTION
**Pull request by The-Marvex**
Successfully compiles with Oracle Java 11

### CODE REVIEW

File: distribution/build-configuration/version.info [hunk 1]
```diff
@@ -1,5 +1,5 @@
-#Thu, 10 Jan 2019 15:15:21 -0500
+#Fri, 01 Mar 2019 03:16:49 +0530
 bindaas.framework.version.major=3
 bindaas.framework.version.minor=3
-bindaas.framework.version.revision=8
-bindaas.build.date=201901101515
+bindaas.framework.version.revision=9
+bindaas.build.date=201903010316
```
Comment (pradeeban) [inline distribution/build-configuration/version.info]: Avoid committing this file. It auto-increments. But we commit this file only when we increment the version.

File: pom.xml [hunk 2]
```diff
@@ -11,14 +11,18 @@
...
         <pluginManagement>
             <plugins>
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-surefire-plugin</artifactId>
-                    <version>2.19.1</version>
+                    <version>2.22.0</version>
+                    <configuration>
+                        <argLine>
+                            --illegal-access=permit
```
Comment (pradeeban) [inline pom.xml]: Can you explain why did you have to set this flag?

File: pom.xml [hunk 3]
```diff
@@ -153,6 +153,63 @@
...
+		    <artifactId>javax.transaction-api</artifactId>
+		    <version>1.2</version>
+		</dependency>
+		<dependency>
+		    <groupId>javax.annotation</groupId>
+		    <artifactId>javax.annotation-api</artifactId>
+		    <version>1.3.1</version>
+		</dependency>
+		<dependency>
+  <groupId>javax.xml.bind</groupId>
```
Comment (pradeeban) [inline pom.xml]: Pls fix the indentation of these lines.

File: pom.xml [hunk 4]
```diff
@@ -153,6 +153,63 @@
...
+		    <artifactId>javax.activation</artifactId>
+		    <version>1.2.0</version>
+		</dependency>
+		<dependency>
+		    <groupId>javax.transaction</groupId>
+		    <artifactId>javax.transaction-api</artifactId>
+		    <version>1.2</version>
+		</dependency>
+		<dependency>
+		    <groupId>javax.annotation</groupId>
```
Comment (pradeeban) [inline pom.xml]: Why did you have to add all these dependencies? Any reason?

File: pom.xml [hunk 5]
```diff
@@ -153,6 +153,63 @@
...
+		<dependency>
+  <groupId>javax.xml.bind</groupId>
+  <artifactId>jaxb-api</artifactId>
+  <version>2.3.1</version>
+</dependency>
+
+<dependency>
+  <groupId>org.glassfish.jaxb</groupId>
+  <artifactId>jaxb-runtime</artifactId>
+  <version>2.3.1</version>
```
Comment (pradeeban) [inline pom.xml]: Why these additional dependencies?

File: projects/core/bindaas-psuedo-sts/pom.xml [hunk 6]
```diff
@@ -37,5 +39,22 @@
...
+			<groupId>org.glassfish.jaxb</groupId>
+			<artifactId>jaxb-runtime</artifactId>
+			<version>${jaxb-api.version}</version>
+		</dependency>
+		<!-- JAXB needs javax.activation module (jdk9) -->
+		<dependency>
+			<groupId>javax.activation</groupId>
+			<artifactId>javax.activation-api</artifactId>
+			<version>1.2.0</version>
+		</dependency>
```
Comment (pradeeban) [inline projects/core/bindaas-psuedo-sts/pom.xml]: Why these dependencies here?

File: projects/core/bindaas-psuedo-sts/src/main/java/edu/emory/cci/bindaas/pseudosts/impl/PseduoSecurityTokenServiceImpl.java [hunk 7]
```diff
@@ -3,12 +3,12 @@
 import java.util.Collection;
 import java.util.Dictionary;
 import java.util.Hashtable;
-
 import javax.ws.rs.GET;
 import javax.ws.rs.HeaderParam;
 import javax.ws.rs.Path;
 import javax.ws.rs.QueryParam;
 import javax.ws.rs.core.Response;
+import javax.xml.*;
```
Comment (pradeeban) [inline projects/core/bindaas-psuedo-sts/src/main/java/edu/emory/cci/bindaas/pseudosts/impl/PseduoSecurityTokenServiceImpl.java]: Looks to me like an unnecessary import. Pls remove and pls remove the file from the commit.

File: projects/core/bindaas-trusted-application-manager/pom.xml [hunk 8]
```diff
@@ -36,6 +36,10 @@
 			<version>0.0.1-SNAPSHOT</version>
 			<artifactId>ldap-bindaas-authentication-provider</artifactId>
 		</dependency>
+        <dependency>
+            <groupId>javax.xml.bind</groupId>
+            <artifactId>jaxb-api</artifactId>
+        </dependency>
```
Comment (pradeeban) [inline projects/core/bindaas-trusted-application-manager/pom.xml]: Why do we need this dependency?

File: projects/core/bindaas-trusted-application-manager/src/main/java/edu/emory/cci/bindaas/trusted_app/impl/TrustedApplicationManagerImpl.java [hunk 9]
```diff
@@ -12,7 +12,7 @@
 import javax.ws.rs.Path;
 import javax.ws.rs.QueryParam;
 import javax.ws.rs.core.Response;
-import javax.xml.bind.DatatypeConverter;
+import javax.xml.bind.*;
```
Comment (pradeeban) [inline projects/core/bindaas-trusted-application-manager/src/main/java/edu/emory/cci/bindaas/trusted_app/impl/TrustedApplicationManagerImpl.java]: Don't unnecessarily load the whole bunch of classes when you can just load one! i.e., this change was unnecessary (unless you can explain).

File: projects/core/bindaas-trusted-application-manager/src/main/java/edu/emory/cci/bindaas/trusted_app/impl/TrustedApplicationManagerImpl.java [hunk 10]
```diff
@@ -246,7 +246,7 @@ public static String calculateDigestValue(String applicationID,
 		String prenonce = roundoff + "|" + applicationKey;
 		byte[] nonceBytes = MessageDigest.getInstance("SHA-1").digest(
 				prenonce.getBytes("UTF-8"));
-		String nonce = DatatypeConverter.printBase64Binary(nonceBytes);
+		String nonce = javax.xml.bind.DatatypeConverter.printBase64Binary(nonceBytes);
```
Comment (pradeeban) [inline projects/core/bindaas-trusted-application-manager/src/main/java/edu/emory/cci/bindaas/trusted_app/impl/TrustedApplicationManagerImpl.java]: Please revert this line break.

File: projects/plugins/data_providers/aime/aime4-bindaas-datasource-provider/pom.xml [hunk 11]
```diff
@@ -67,6 +67,10 @@
 			<artifactId>bindaas-core-api</artifactId>
 			<version>0.0.1</version>
 		</dependency>
-	</dependencies>
+        <dependency>
+            <groupId>javax.xml.bind</groupId>
+            <artifactId>jaxb-api</artifactId>
```
Comment (pradeeban) [inline projects/plugins/data_providers/aime/aime4-bindaas-datasource-provider/pom.xml]: Why do we need this dependency here?

File: projects/plugins/data_providers/aime/aime4-bindaas-datasource-provider/src/main/java/edu/emory/cci/bindaas/datasource/provider/aime4/jaxb/EDDocRef.java [hunk 12]
```diff
@@ -13,11 +13,12 @@
 import javax.xml.bind.annotation.XmlType;
 
 
+
 /**
```
Comment (pradeeban) [inline projects/plugins/data_providers/aime/aime4-bindaas-datasource-provider/src/main/java/edu/emory/cci/bindaas/datasource/provider/aime4/jaxb/EDDocRef.java]: Pls don't change the styles of the classes in the pull request. It unnecessarily adds additional files to the review list, thus making the review a harder task.

File: projects/plugins/data_providers/aime/aime4-bindaas-datasource-provider/src/main/java/edu/emory/cci/bindaas/datasource/provider/aime4/jaxb/EDDocRef.java [hunk 13]
```diff
@@ -13,11 +13,12 @@
...
  * <p>Java class for ED.Doc.Ref complex type.
  * 
  * <p>The following schema fragment specifies the expected content contained within this class.
- * 
+ *
```
Comment (pradeeban) [inline projects/plugins/data_providers/aime/aime4-bindaas-datasource-provider/src/main/java/edu/emory/cci/bindaas/datasource/provider/aime4/jaxb/EDDocRef.java]: As above. Avoid style changes pls.

File: projects/plugins/transformers/blob-download-plugin/pom.xml [hunk 14]
```diff
@@ -137,5 +137,9 @@
 			<artifactId>javax.json</artifactId>
 			<version>1.0.4</version>
 		</dependency>
-	</dependencies>
+        <dependency>
+            <groupId>javax.activation</groupId>
+            <artifactId>activation</artifactId>
+        </dependency>
```
Comment (pradeeban) [inline projects/plugins/transformers/blob-download-plugin/pom.xml]: I don't think this dependency is necessary here. Did you really test the blob-download-plugin?

File: projects/plugins/transformers/generic-output-chainer-query-result-modifier/pom.xml [hunk 15]
```diff
@@ -102,5 +102,9 @@
 			<artifactId>bindaas-core-api</artifactId>
 			<version>0.0.1</version>
 		</dependency>
-	</dependencies>
+        <dependency>
+            <groupId>javax.xml.bind</groupId>
+            <artifactId>jaxb-api</artifactId>
```
Comment (pradeeban) [inline projects/plugins/transformers/generic-output-chainer-query-result-modifier/pom.xml]: Again, probably an unnecessary dependency.

File: distribution/build-configuration/version.info (hunk 1 above)
Comment (The-Marvex) [reply distribution/build-configuration/version.info]: Sure.

File: pom.xml (hunk 4 above)
Comment (The-Marvex) [reply pom.xml]: The major difference while going from Java 8 to Java 11 is the dependencies that were removed, this is one of those.

File: projects/plugins/data_providers/aime/aime4-bindaas-datasource-provider/pom.xml (hunk 11 above)
Comment (The-Marvex) [reply projects/plugins/data_providers/aime/aime4-bindaas-datasource-provider/pom.xml]: same here as well.

File: pom.xml (hunk 5 above)
Comment (The-Marvex) [reply pom.xml]: this too.

File: projects/core/bindaas-trusted-application-manager/src/main/java/edu/emory/cci/bindaas/trusted_app/impl/TrustedApplicationManagerImpl.java (hunk 9 above)
Comment (The-Marvex) [reply projects/core/bindaas-trusted-application-manager/src/main/java/edu/emory/cci/bindaas/trusted_app/impl/TrustedApplicationManagerImpl.java]: This is my bad.

File: pom.xml (hunk 4 above)
Comment (pradeeban) [reply pom.xml]: I don't understand your reply. You are adding more dependencies. Not removing any.

Besides, from your message, I understand you haven't managed to get Bindaas running. Please submit pull requests only after confirming it does not break any features.

Comment (The-Marvex) [reply pom.xml]: I was able to get bindaas running after making all the changes.

Comment (The-Marvex) [reply pom.xml]: The dependency was removed in Java 11. I added that dependency which is still available for Java 11. I'll create a PR in dev branch for you to check

**Pull request closed without merging**